rustdoc-search: use lowercase, non-normalized name for type search

notriddle opened the pull request:
The type name ID map has underscores in its names, so the query element should have them, too.

Fixes #125993

fmease commented:
@bors r+

bors commented:
:pushpin: Commit ff0c10c37511ce86379c0e6223c224cc2f0b9ba0 has been approved by `fmease`

It is now in the [queue](https://bors.rust-lang.org/homu/queue/rust) for this repository.

<!-- @bors r=fmease ff0c10c37511ce86379c0e6223c224cc2f0b9ba0 -->
<!-- homu: {"type":"Approved","sha":"ff0c10c37511ce86379c0e6223c224cc2f0b9ba0","approver":"fmease","queue":"https://bors.rust-lang.org/homu/queue/rust"} -->

bors commented:
:hourglass: Testing commit ff0c10c37511ce86379c0e6223c224cc2f0b9ba0 with merge a47d1e2e686cff3141f9cc33e2b782995d6ce93a...
<!-- homu: {"type":"BuildStarted","head_sha":"ff0c10c37511ce86379c0e6223c224cc2f0b9ba0","merge_sha":"a47d1e2e686cff3141f9cc33e2b782995d6ce93a"} -->

rust-log-analyzer commented:
The job **`aarch64-apple`** failed! Check out the build log: [(web)](https://github.com/rust-lang-ci/rust/actions/runs/9434352483/job/25986362712) [(plain)](https://github.com/rust-lang-ci/rust/commit/a47d1e2e686cff3141f9cc33e2b782995d6ce93a/checks/25986362712/logs)

<details><summary><i>Click to see the possible cause of the failure (guessed by this bot)</i></summary>

```plain
Testing parser-tuple.js ... OK
Testing parser-weird-queries.js ... OK
Testing path-maxeditdistance.js ... OK
Testing path-ordering.js ... OK
Testing pidt.js ... FAILED
[ query `pid_t`]==> Result not found in 'returned': '{"path":"std::os::unix::net::SocketCred","name":"get_pid"}'
 {
-     "path": "std::os::unix::net::SocketCred",
-     "path": "std::os::unix::net::SocketCred",
-     "name": "get_pid",
Testing primitive.js ... OK
Testing println-typo.js ... OK
Testing quoted.js ... OK
Testing reference-shrink.js ... OK
```

</details>

bors commented:
:broken_heart: Test failed - [checks-actions](https://github.com/rust-lang-ci/rust/actions/runs/9434352483/job/25987008788)
<!-- homu: {"type":"BuildFailed","builder_url":"https://github.com/rust-lang-ci/rust/actions/runs/9434352483/job/25987008788","builder_name":"checks-actions"} -->

fmease commented:
Right, I guess we make this a platform-agnostic test ^^'.

Min:

```rs
pub type pid_t = i32;
pub fn get_pid() -> pid_t { 0 }
pub fn set_pid(_: pid_t) {}
```

bors commented:
:pushpin: Commit f2d707dabde9c89dfe0b1c728dc0c686b8732d6a has been approved by `fmease`

It is now in the [queue](https://bors.rust-lang.org/homu/queue/rust) for this repository.

<!-- @bors r=fmease f2d707dabde9c89dfe0b1c728dc0c686b8732d6a -->
<!-- homu: {"type":"Approved","sha":"f2d707dabde9c89dfe0b1c728dc0c686b8732d6a","approver":"fmease","queue":"https://bors.rust-lang.org/homu/queue/rust"} -->

rust-log-analyzer commented:
The job **`mingw-check-tidy`** failed! Check out the build log: [(web)](https://github.com/rust-lang/rust/actions/runs/9438356042/job/25995240747) [(plain)](https://github.com/rust-lang/rust/commit/f2d707dabde9c89dfe0b1c728dc0c686b8732d6a/checks/25995240747/logs)

<details><summary><i>Click to see the possible cause of the failure (guessed by this bot)</i></summary>

```plain
Getting action download info
Download action repository 'msys2/setup-msys2@v2.22.0' (SHA:cc11e9188b693c2b100158c3322424c4cc1dadea)
Download action repository 'actions/checkout@v4' (SHA:a5ac7e51b41094c92402da3b24376905380afc29)
Download action repository 'actions/upload-artifact@v4' (SHA:65462800fd760344b1a7b4382951275a0abb4808)
Complete job name: PR - mingw-check-tidy
git config --global core.autocrlf false
shell: /usr/bin/bash --noprofile --norc -e -o pipefail {0}
---
COPY scripts/sccache.sh /scripts/
RUN sh /scripts/sccache.sh

COPY host-x86_64/mingw-check/reuse-requirements.txt /tmp/
RUN pip3 install --no-deps --no-cache-dir --require-hashes -r /tmp/reuse-requirements.txt \
    && pip3 install virtualenv
COPY host-x86_64/mingw-check/validate-toolstate.sh /scripts/
COPY host-x86_64/mingw-check/validate-error-codes.sh /scripts/

# NOTE: intentionally uses python2 for x.py so we can test it still works.
# NOTE: intentionally uses python2 for x.py so we can test it still works.
# validate-toolstate only runs in our CI, so it's ok for it to only support python3.
ENV SCRIPT TIDY_PRINT_DIFF=1 python2.7 ../x.py test \
           --stage 0 src/tools/tidy tidyselftest --extra-checks=py:lint
# This file is autogenerated by pip-compile with Python 3.10
# by the following command:
#
#    pip-compile --allow-unsafe --generate-hashes reuse-requirements.in
---

#12 [5/8] COPY host-x86_64/mingw-check/reuse-requirements.txt /tmp/
#12 DONE 0.0s

#13 [6/8] RUN pip3 install --no-deps --no-cache-dir --require-hashes -r /tmp/reuse-requirements.txt     && pip3 install virtualenv
#13 0.419   Downloading binaryornot-0.4.4-py2.py3-none-any.whl (9.0 kB)
#13 0.431 Collecting boolean-py==4.0
#13 0.435   Downloading boolean.py-4.0-py3-none-any.whl (25 kB)
#13 0.446 Collecting chardet==5.1.0
---
#13 3.460 Building wheels for collected packages: reuse
#13 3.461   Building wheel for reuse (pyproject.toml): started
#13 3.788   Building wheel for reuse (pyproject.toml): finished with status 'done'
#13 3.790   Created wheel for reuse: filename=reuse-1.1.0-cp310-cp310-manylinux_2_35_x86_64.whl size=181117 sha256=f5f58750481f69515c2c0d1d503daf565e2565c370d07fc6aeb95fe3498b4269
#13 3.790   Stored in directory: /tmp/pip-ephem-wheel-cache-yjgr6tkl/wheels/c2/3c/b9/1120c2ab4bd82694f7e6f0537dc5b9a085c13e2c69a8d0c76d
#13 3.793 Installing collected packages: boolean-py, binaryornot, setuptools, reuse, python-debian, markupsafe, license-expression, jinja2, chardet
#13 3.815   Attempting uninstall: setuptools
#13 3.816     Found existing installation: setuptools 59.6.0
#13 3.817     Not uninstalling setuptools at /usr/lib/python3/dist-packages, outside environment /usr
---
#13 5.069   Downloading virtualenv-20.26.2-py3-none-any.whl (3.9 MB)
#13 5.148      ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 3.9/3.9 MB 50.5 MB/s eta 0:00:00
#13 5.200 Collecting filelock<4,>=3.12.2
#13 5.204   Downloading filelock-3.14.0-py3-none-any.whl (12 kB)
#13 5.220 Collecting distlib<1,>=0.3.7
#13 5.226   Downloading distlib-0.3.8-py2.py3-none-any.whl (468 kB)
#13 5.232      ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 468.9/468.9 KB 109.2 MB/s eta 0:00:00
#13 5.262 Collecting platformdirs<5,>=3.9.1
#13 5.275   Downloading platformdirs-4.2.2-py3-none-any.whl (18 kB)
#13 5.361 Installing collected packages: distlib, platformdirs, filelock, virtualenv
#13 5.525 Successfully installed distlib-0.3.8 filelock-3.14.0 platformdirs-4.2.2 virtualenv-20.26.2
#13 DONE 5.6s

#14 [7/8] COPY host-x86_64/mingw-check/validate-toolstate.sh /scripts/
#14 DONE 0.0s
---
DirectMap4k:      192448 kB
DirectMap2M:     8196096 kB
DirectMap1G:    10485760 kB
##[endgroup]
Executing TIDY_PRINT_DIFF=1 python2.7 ../x.py test            --stage 0 src/tools/tidy tidyselftest --extra-checks=py:lint
+ TIDY_PRINT_DIFF=1 python2.7 ../x.py test --stage 0 src/tools/tidy tidyselftest --extra-checks=py:lint
    Finished `dev` profile [unoptimized] target(s) in 0.03s
##[endgroup]
downloading https://ci-artifacts.rust-lang.org/rustc-builds-alt/7bb0ef4902e43f338aa73c3ff728bea3cb9ecf6a/rust-dev-nightly-x86_64-unknown-linux-gnu.tar.xz
extracting /checkout/obj/build/cache/llvm-7bb0ef4902e43f338aa73c3ff728bea3cb9ecf6a-true/rust-dev-nightly-x86_64-unknown-linux-gnu.tar.xz to /checkout/obj/build/x86_64-unknown-linux-gnu/ci-llvm
---
   Compiling tidy v0.1.0 (/checkout/src/tools/tidy)
    Finished `release` profile [optimized] target(s) in 27.92s
##[endgroup]
fmt check
##[error]Diff in /checkout/tests/rustdoc-js/underscoredtype.rs at line 1:
 pub mod unix {
- #[allow(non_camel_case_types)]
- pub type pid_t = i32;
- pub fn get_pid() -> pid_t { 0 }
- pub fn set_pid(_: pid_t) {}
+    pub type pid_t = i32;
+    pub type pid_t = i32;
+    pub fn get_pid() -> pid_t {
+    }
+    }
+    pub fn set_pid(_: pid_t) {}
 
 
fmt error: Running `"/checkout/obj/build/x86_64-unknown-linux-gnu/rustfmt/bin/rustfmt" "--config-path" "/checkout" "--edition" "2021" "--unstable-features" "--skip-children" "--check" "/checkout/tests/rustdoc-js/assoc-type-backtrack.rs" "/checkout/tests/rustdoc-js/search-method-disambiguate.rs" "/checkout/tests/rustdoc-js/macro-search.rs" "/checkout/tests/rustdoc-js/doc-alias-whitespace.rs" "/checkout/tests/rustdoc-js/search-bag-semantics.rs" "/checkout/tests/rustdoc-js/raw-pointer.rs" "/checkout/tests/rustdoc-js/search-short-types.rs" "/checkout/tests/rustdoc-js/auxiliary/interner.rs" "/checkout/tests/rustdoc-js/auxiliary/macro-in-module.rs" "/checkout/tests/rustdoc-js/auxiliary/equivalent.rs" "/checkout/tests/rustdoc-js/looks-like-rustc-interner.rs" "/checkout/tests/rustdoc-js/underscoredtype.rs" "/checkout/tests/rustdoc-js/generics-trait.rs" "/checkout/tests/rustdoc-js/doc-alias-filter-out.rs" "/checkout/tests/rustdoc-js/reexport-dedup.rs" "/checkout/tests/rustdoc-js/full-path-function.rs" "/checkout/tests/rustdoc-js/foreign-type-path.rs" "/checkout/tests/rustdoc-js/reference.rs" "/checkout/tests/rustdoc-js/generics-match-ambiguity.rs" "/checkout/tests/rustdoc-js/non-english-identifier.rs" "/checkout/compiler/rustc_codegen_ssa/src/back/lto.rs" "/checkout/compiler/rustc_codegen_ssa/src/back/metadata.rs" "/checkout/compiler/rustc_codegen_ssa/src/back/mod.rs" "/checkout/compiler/rustc_codegen_ssa/src/back/link.rs" "/checkout/compiler/rustc_codegen_ssa/src/back/rpath/tests.rs" "/checkout/compiler/rustc_codegen_ssa/src/back/write.rs" "/checkout/compiler/rustc_codegen_ssa/src/back/command.rs" "/checkout/compiler/rustc_codegen_ssa/src/back/linker.rs" "/checkout/compiler/rustc_codegen_ssa/src/back/rpath.rs" "/checkout/compiler/rustc_codegen_ssa/src/back/archive.rs" "/checkout/compiler/rustc_codegen_ssa/src/back/symbol_export.rs" "/checkout/compiler/rustc_codegen_ssa/src/lib.rs" "/checkout/compiler/rustc_codegen_ssa/src/assert_module_sources.rs" "/checkout/compiler/rustc_codegen_ssa/src/debuginfo/mod.rs" "/checkout/compiler/rustc_codegen_ssa/src/debuginfo/type_names.rs" "/checkout/compiler/rustc_codegen_ssa/src/traits/builder.rs" "/checkout/compiler/rustc_codegen_ssa/src/traits/mod.rs" "/checkout/compiler/rustc_codegen_ssa/src/traits/misc.rs" "/checkout/compiler/rustc_codegen_ssa/src/traits/intrinsic.rs" "/checkout/compiler/rustc_codegen_ssa/src/traits/write.rs" "/checkout/compiler/rustc_codegen_ssa/src/traits/statics.rs" "/checkout/compiler/rustc_codegen_ssa/src/traits/abi.rs" "/checkout/compiler/rustc_codegen_ssa/src/traits/coverageinfo.rs" "/checkout/compiler/rustc_codegen_ssa/src/traits/asm.rs" "/checkout/compiler/rustc_codegen_ssa/src/traits/declare.rs" "/checkout/compiler/rustc_codegen_ssa/src/traits/backend.rs" "/checkout/compiler/rustc_codegen_ssa/src/traits/debuginfo.rs" "/checkout/compiler/rustc_codegen_ssa/src/traits/consts.rs" "/checkout/compiler/rustc_codegen_ssa/src/traits/type_.rs" "/checkout/compiler/rustc_codegen_ssa/src/mir/mod.rs" "/checkout/compiler/rustc_codegen_ssa/src/mir/rvalue.rs" "/checkout/compiler/rustc_codegen_ssa/src/mir/intrinsic.rs" "/checkout/compiler/rustc_codegen_ssa/src/mir/operand.rs" "/checkout/compiler/rustc_codegen_ssa/src/mir/block.rs" "/checkout/compiler/rustc_codegen_ssa/src/mir/constant.rs" "/checkout/compiler/rustc_codegen_ssa/src/mir/coverageinfo.rs" "/checkout/compiler/rustc_codegen_ssa/src/mir/locals.rs" "/checkout/compiler/rustc_codegen_ssa/src/mir/place.rs" "/checkout/compiler/rustc_codegen_ssa/src/mir/debuginfo.rs" "/checkout/compiler/rustc_codegen_ssa/src/mir/analyze.rs" "/checkout/compiler/rustc_codegen_ssa/src/mir/statement.rs" "/checkout/compiler/rustc_codegen_ssa/src/errors.rs" "/checkout/compiler/rustc_codegen_ssa/src/target_features.rs" "/checkout/tests/rustdoc-js/never-search.rs"` failed.
If you're running `tidy`, try again with `--bless`. Or, if you just want to format code, run `./x.py fmt` instead.
  local time: Sun Jun  9 17:40:19 UTC 2024
  network time: Sun, 09 Jun 2024 17:40:20 GMT
##[error]Process completed with exit code 1.
Post job cleanup.
```

</details>

jieyouxu commented:
Unfortunately this needs to `x fmt`.
@bors r-

notriddle committed:
rustdoc-search: use lowercase, non-normalized name for type search

The type name ID map has underscores in its names, so the query
element should have them, too.

bors commented:
:pushpin: Commit 8865b8c63900acb1834390344fc1cc43ae209f08 has been approved by `fmease`

It is now in the [queue](https://bors.rust-lang.org/homu/queue/rust) for this repository.

<!-- @bors r=fmease 8865b8c63900acb1834390344fc1cc43ae209f08 -->
<!-- homu: {"type":"Approved","sha":"8865b8c63900acb1834390344fc1cc43ae209f08","approver":"fmease","queue":"https://bors.rust-lang.org/homu/queue/rust"} -->
